src/pages/workspaces/workspace/jobHistory/WorkflowDashboard.js

```diff
@@ -0,0 +1,269 @@
+        ]): undefined,
+        failed ? tr({}, [
+          td(styles.statusDetailCell, [failedIcon()]),
+          td(['Failed']),
+          td(styles.statusDetailCell, [failed])
+        ]): undefined,
+        _.map(other => tr({}, [
+          td(styles.statusDetailCell, [unknownIcon()]),
+          td([other]),
+          td(styles.statusDetailCell, [others[other]])
+        ]), Object.keys(others))
+      ].filter(element => element !== undefined))
+    ])
+
+  ])
+}
+
+const WorkflowDashboard = _.flow(
+  Utils.forwardRefWithName('WorkflowDashboard'),
+  wrapWorkspace({
+    breadcrumbs: props => breadcrumbs.commonPaths.workspaceDashboard(props),
+    title: 'Job History', activeTab: 'job history'
+  })
+)((props, ref) => {
+  const { namespace, name, submissionId, workflowId, workspace: { workspace: { bucketName } } } = props
+
+  /*
+   * State setup
+   */
+  const [workflow, setWorkflow] = useState({})
+  const [showLog, setShowLog] = useState(false)
+  const [failureCopied, setFailureCopied] = useState()
+
+  const signal = Utils.useCancellation()
+
+  /*
+   * Data fetchers
+   */
+
+  useEffect(() => {
+    const initialize = withErrorReporting('Unable to fetch Workflow Details',
+      async () => {
+        // If the workflow is empty, or we need to refresh after 60s:
+        if (_.isEmpty(workflow) || _.includes(workflow.status, ['Running', 'Submitted'])) {
+          if (!_.isEmpty(workflow)) {
+            await Utils.delay(60000)
+          }
+          const includeKey = [
+            'backendLogs',
+            'backendStatus',
+            'end',
+            'executionStatus',
+            'callCaching:hit',
+            'failures',
+            'id',
+            'jobId',
+            'start',
+            'status',
+            'stderr',
+            'stdout',
+            'submission',
+            'submittedFiles:workflow',
+            'subworkflowId',
+            'workflowLog',
+            'workflowName',
+            'workflowRoot'
+          ]
+          const wf = await Ajax(signal).Workspaces.workspace(namespace, name).submission(submissionId).getWorkflow(workflowId, includeKey)
+          setWorkflow(wf)
+        }
+      })
+
+    initialize()
+  }, [workflow]) // eslint-disable-line react-hooks/exhaustive-deps
+
+  /*
+   * Page render
+   */
+  const {
+    end,
+    failures,
+    start,
+    status,
+    workflowLog,
+    workflowName,
+    submittedFiles: { workflow: wdl } = {}
+  } = workflow
+
+  const restructureFailures = _.map(({ message, causedBy }) => ({
+    message,
+    ...(!_.isEmpty(causedBy) ? { causedBy: simplifyDidNotStartFailures(restructureFailures(causedBy)) } : {})
+  }))
+
+  const simplifyDidNotStartFailures = failuresArray => {
+    const filtered = _.filter(({ message }) => !_.isEmpty(message) && !message.startsWith('Will not start job'), failuresArray)
+    const sizeDiff = !_.isEmpty(failuresArray) ? failuresArray.length - filtered.length : 0
```

Comment:
This guard is unnecessary, since `filtered` is filtering `failuresArray`, so it'll always be the same length or shorter.
```suggestion
    const sizeDiff = failuresArray.length - filtered.length
```

Reply:
This guard is to catch the case where `failuresArray` is undefined and thus calling `.length` on it would be an error.

Reply:
What if we defaulted it to an empty array where we declare the arg, then?

Reply:
It looks like this function and the one above are always called together. Perhaps we could combine them for clarity?